appcues/src/main/java/com/appcues/data/AppcuesRepository.kt

```diff
@@ -148,13 +150,21 @@ internal class AppcuesRepository(
 
             qualifyResult.doIfSuccess { response ->
                 val priority: ExperiencePriority = if (response.qualificationReason == "screen_view") LOW else NORMAL
-                experiences += response.experiences.map { experienceMapper.map(it, priority, response.experiments, activity.requestId) }
+                experiences += response.experiences.filterIsInstance<ExperienceResponse>().map {
+                    experienceMapper.map(it, priority, response.experiments, activity.requestId)
+                }
             }
 
             qualifyResult.doIfFailure {
                 logcues.info("qualify request failed, reason: $it")
                 when (it) {
-                    is NetworkError -> successful = false
+                    is NetworkError -> {
+                        when (it.throwable) {
+                            // don't retry on JSON parse failures, server responded, it was just unexpected structure
+                            is JsonDataException -> Unit
+                            else -> successful = false
+                        }
+                    }
```

Comment:
not directly related to this PR feature, but an issue i noticed - we should not retry _every_ `NetworkError`. I noticed that this code path is hit when a valid server response comes back, but we just cannot deserialize it - the `JsonDataException`.  This type of exception should not trigger a re-POST to the server, as the server has already received and processed this request.